### DISCUSSION TITLE
GitHub Action not working

### DISCUSSION THREAD

**Discussion by benmwatson** [philosowaffle/peloton-to-garmin#417] [General]
Here's the log, I think it should be public, right? https://github.com/benmwatson/peloton-to-garmin/actions/runs/3933212271/jobs/6726672680

It looks like it's not even running 

The one secret I didn't create was the DEVICE_INFO because the wiki page for it said it was optional, and I don't think I created it before when I was using Docker. But is DEVICE_INFO required for the GitHub action version? I'm trying to move to the GitHub solution so I can avoid needing to run it on my personal PC. 

If I do need to create a DEVICE_INFO file, how do I go about creating it. It says to use a TCX file, but where would that be created?

I've also attached the raw log here:

[p2glog.txt](https://github.com/philosowaffle/peloton-to-garmin/files/10428708/p2glog.txt)

**Comment by philosowaffle** [comment]
Its most likely skipping the final run command because [this check](https://github.com/benmwatson/peloton-to-garmin/actions/runs/3919699851/workflow#L78) is not being satisfied. I'd double check the spelling and number of underscores (single and double) used when setting up the secret names.

Edit: To answer your other question, no, you do not need DeviceInfo I don't think its the reason this job is not running. But for reference the steps for creating a DeviceInfo file are documented [here](https://philosowaffle.github.io/peloton-to-garmin/configuration/providing-device-info.html#custom-device-info).

**Reply by benmwatson** [reply]
Oh man...now I feel like an idiot. It was the double underscores I missed. Should have copied and pasted the variable names instead of just eyeball it.  Thank you so much. All working now.
